File: Data Structures & Algorithms/add-two-numbers/submission-4.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        // Brute force, convert to array operate > convert to linked list
        ListNode* dummy = new ListNode(0);
        ListNode* curr = dummy; 
        int carry = 0;
        while(l1 || l2 || carry!=0){
            int val1 = (l1!=nullptr) ? l1->val : 0;
            int val2 = (l2!=nullptr) ? l2->val : 0;

            int sum = val1 + val2 + carry;
            carry = sum/10;
            int digit = sum % 10;

            curr -> next = new ListNode(digit);
            curr = curr->next;
            if(l1){
                l1 = l1->next;
            }
            if(l2){
                l2 = l2->next;
            }
        }
        ListNode* head = dummy->next;
        delete dummy;
        return head;
    }
};

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-4.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        prev = None
        curr = head

        while n!=0:
            curr = curr.next
            n -= 1

        while curr!=None:
            curr = curr.next
            if prev:
                prev = prev.next
            else:
                prev = head
        if prev==None:
            if head.next:
                return head.next
            else:
                return None
        
        prev.next = prev.next.next
        return head